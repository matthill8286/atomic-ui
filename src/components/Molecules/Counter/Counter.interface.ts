export interface CounterProps extends CounterValues, CounterAppearance {}

export type CounterSize = 'sm' | 'lg'

export interface CounterValues {
  days: number
  hours: number
  minutes: number
  seconds: number
}
export interface TimestampsProps {
  from?: Date
  toUTC?: string
}
export interface CounterAppearance {
  showSeconds?: boolean
  maxSize?: CounterSize
  colors?: CounterItemColors
  outline?: boolean
  lastDayShowSecondsHideDays?: boolean
  maxDaysRange?: number
  loading?: boolean
}

export interface CounterItemColors {
  fontColor?: string
  circleColor?: string
  progressColor?: string
  fill?: string
}
