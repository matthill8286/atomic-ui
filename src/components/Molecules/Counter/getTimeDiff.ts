export interface TimeDiff {
  days: number
  hours: number
  minutes: number
  seconds: number
}

export const getTimeDiff = (start: Date, end: Date): TimeDiff => {
  const diffInSeconds = Math.floor((end.getTime() - start.getTime()) / 1000)
  if (diffInSeconds <= 0) {
    return { days: 0, hours: 0, minutes: 0, seconds: 0 }
  }
  return {
    days: Math.floor(diffInSeconds / 86400),
    hours: Math.floor((diffInSeconds / 3600) % 24),
    minutes: Math.floor((diffInSeconds / 60) % 60),
    seconds: Math.floor(diffInSeconds % 60),
  }
}
