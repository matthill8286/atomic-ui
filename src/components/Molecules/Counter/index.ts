export * from './Counter'
export * from './CounterWithTimer'
export * from './Counter.hook'
