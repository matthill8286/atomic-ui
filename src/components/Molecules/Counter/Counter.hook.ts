import * as React from 'react'
import { useInterval } from '@/utils'
import { TimestampsProps } from './Counter.interface'
import { getTimeDiff, TimeDiff } from './getTimeDiff'

export function useCounter({ from = new Date(), toUTC }: TimestampsProps) {
  const initToDate = toUTC ? new Date(parseDate(toUTC)) : new Date(new Date().getTime())

  const [timeDiffValues, setTimeDiffValues] = React.useState<TimeDiff>(
    getTimeDiff(from, initToDate)
  )
  const [loading, setLoading] = React.useState(true)

  React.useEffect(() => {
    if (toUTC) {
      setTimeDiffValues(getTimeDiff(from, new Date(parseDate(toUTC))))
      setLoading(false)
    } else {
      setLoading(true)
    }
  }, [toUTC])

  useInterval(() => {
    if (toUTC) {
      setTimeDiffValues(getTimeDiff(from, new Date(parseDate(toUTC))))
    }
  }, 1000)
  return { timeDiffValues, loading }
}

function parseDate(date: string) {
  const parsed = Date.parse(date)
  if (!isNaN(parsed)) {
    return parsed
  }

  return Date.parse(date.replace(/-/g, '/').replace(/[a-z]+/gi, ' '))
}
