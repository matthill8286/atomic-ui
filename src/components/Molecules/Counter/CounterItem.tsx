import React, { useEffect, useState } from 'react'
import { Typo } from '@/components/Atoms/Typography'
import { styled } from '@/styles/styled'
import { useInterval } from '@/utils/useInterval'
import { CounterItemColors, CounterSize } from './Counter.interface'

const STROKE_WIDTH_LG = 8
const STROKE_WIDTH_SM = 4
const OUTLINE_OFFSET = 4

const MAX_RADIUS_LG = 164
const MAX_RADIUS_SM = 96
const MIN_RADIUS_SM = 85

const COUNTER_VALUE_FONT_SIZE_LG = '80px'
const COUNTER_VALUE_FONT_SIZE_SM = '32px'

const COUTER_VALUE_OFFSET_TOP_LG = '60%'
const COUTER_VALUE_OFFSET_TOP_SM = '51%'

const COUTER_LABEL_OFFSET_TOP_LG = '-55px'
const COUTER_LABEL_OFFSET_TOP_SM = '-48px'

const StyledCounterItem = styled.div<{ size: CounterSize }>(
  ({ size }) => `
  max-height: ${size === 'sm' ? MAX_RADIUS_SM : MAX_RADIUS_LG}px;
  min-height: ${MIN_RADIUS_SM}px;
  max-width: ${size === 'sm' ? MAX_RADIUS_SM : MAX_RADIUS_LG}px;
  min-width: ${MIN_RADIUS_SM}px;
`
)

const StyledBgCircle = styled.circle<{ color?: string }>(
  ({ color, theme }) => `
  stroke: ${color || theme.color.grey2};
`
)

const StyledProgressCircle = styled.circle<{ color?: string }>(
  ({ color, theme }) => `
  stroke: ${color || theme.color.primary};
`
)

const StyledCounterValue = styled.text<{ size: CounterSize; color?: string }>(
  ({ color, size, theme }) => `
  font-weight: bold;
  font-family: ${theme.font.family.featured};
  fill: ${color || theme.color.grey5};
  font-size: ${size === 'sm' ? COUNTER_VALUE_FONT_SIZE_SM : COUNTER_VALUE_FONT_SIZE_LG};
`
)

const StyledCounterLabelWrapper = styled.div<{ size: CounterSize; color?: string }>(
  ({ color, size, theme }) => `
  position: relative;
  text-align: center;
  top: ${size === 'sm' ? COUTER_LABEL_OFFSET_TOP_SM : COUTER_LABEL_OFFSET_TOP_LG};
  span {
    color: ${color || theme.color.grey4};
    font-size: ${size === 'sm' ? theme.font.size.xxs : theme.font.size.xs};
  }
`
)

interface CounterItemProps {
  maxValue: number
  label: string
  centerValue: number
  size?: CounterSize
  colors?: CounterItemColors
  outline?: boolean
  loading?: boolean
}

export const CounterItem: React.FC<CounterItemProps> = ({
  maxValue,
  label,
  centerValue,
  size = 'lg',
  colors = {},
  outline = false,
  loading = false,
}): JSX.Element => {
  const [percentage, setPercentage] = useState(0)
  const [counter, setCounter] = useState(0)
  const { fill, fontColor, circleColor, progressColor } = colors
  const centerRadius = size === 'sm' ? MAX_RADIUS_SM / 2 : MAX_RADIUS_LG / 2
  const strokeWidth = size === 'sm' ? STROKE_WIDTH_SM : STROKE_WIDTH_LG
  const innerRadiusRegular = centerRadius - strokeWidth / 2
  const innerRadiusOutline = outline ? innerRadiusRegular - OUTLINE_OFFSET : innerRadiusRegular
  const perimeter = Math.PI * innerRadiusOutline * 2
  const strokeDashOffset = Math.round(((100 - Math.min(percentage, 98)) / 100) * perimeter)

  const initProgress = () => {
    return () => {
      if (loading) {
        setPercentage(!percentage ? 100 : 0)
        setCounter(counter + 1)
      } else {
        setPercentage((100 / maxValue) * centerValue)
      }
    }
  }
  useEffect(initProgress(), [])
  useEffect(() => {
    if (!loading) {
      setPercentage((100 / maxValue) * centerValue)
    }
  }, [centerValue, loading])

  useInterval(() => {
    if (loading) {
      setPercentage(!percentage ? 100 : 0)
      setCounter(counter + 1)
    }
  }, 1000)
  return (
    <StyledCounterItem size={size}>
      <svg
        height="100%"
        width="100%"
        viewBox={`${-strokeWidth / 2} ${-strokeWidth / 2} ${centerRadius * 2} ${centerRadius * 2}`}>
        <StyledBgCircle
          cx={innerRadiusRegular}
          cy={innerRadiusRegular}
          r={innerRadiusOutline}
          strokeWidth={strokeWidth + (outline ? OUTLINE_OFFSET * 2 : 0)}
          fill={fill ? fill : 'none'}
          color={circleColor}
        />
        <StyledProgressCircle
          transform={`rotate(${(loading
            ? percentage
              ? 360 * counter
              : (counter + 1) * 360
            : 360 * counter) + 270} ${
            outline ? innerRadiusOutline + OUTLINE_OFFSET : innerRadiusOutline
          } ${outline ? innerRadiusOutline + OUTLINE_OFFSET : innerRadiusOutline})`}
          cx={outline ? innerRadiusOutline + OUTLINE_OFFSET : innerRadiusOutline}
          cy={outline ? innerRadiusOutline + OUTLINE_OFFSET : innerRadiusOutline}
          r={innerRadiusOutline}
          strokeDasharray={perimeter}
          strokeWidth={strokeWidth + 0.5}
          strokeDashoffset={strokeDashOffset}
          strokeLinecap="round"
          fill="none"
          color={progressColor}
          style={{
            transition: `stroke-dashoffset 1s ease-out${loading ? ', transform 1s linear' : ''}`,
          }}
        />
        {!loading && (
          <StyledCounterValue
            x="50%"
            y={size === 'sm' ? COUTER_VALUE_OFFSET_TOP_SM : COUTER_VALUE_OFFSET_TOP_LG}
            dx={-strokeWidth / 2}
            textAnchor="middle"
            size={size}
            color={fontColor}>
            {centerValue < 10 ? '0' : ''}
            {centerValue}
          </StyledCounterValue>
        )}
      </svg>
      {!loading && (
        <StyledCounterLabelWrapper color={fontColor} size={size}>
          <Typo>{label}</Typo>
        </StyledCounterLabelWrapper>
      )}
    </StyledCounterItem>
  )
}
