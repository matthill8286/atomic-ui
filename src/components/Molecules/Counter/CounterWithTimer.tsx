import * as React from 'react'
import { Counter } from './Counter'
import { useCounter } from './Counter.hook'
import { CounterAppearance, TimestampsProps } from './Counter.interface'

interface CounterWithTimerProps extends TimestampsProps, Omit<CounterAppearance, 'loading'> {}

export const CounterWithTimer: React.FC<CounterWithTimerProps> = ({
  from = new Date(),
  toUTC,
  ...rest
}): JSX.Element => {
  const { loading, timeDiffValues } = useCounter({ from, toUTC })
  return <Counter loading={loading} {...timeDiffValues} {...rest} />
}
