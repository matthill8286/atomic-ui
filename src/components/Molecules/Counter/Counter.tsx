import React from 'react'
import { useWindowDimensions } from '@/components/Helper/useWindowDimensions'
import { breakpoints } from '@/styles/sc-vars-global'
import { styled } from '@/styles/styled'
import { CounterProps, CounterSize } from './Counter.interface'
import { CounterItem } from './CounterItem'

const StyledCounter = styled.div<{ size: CounterSize }>(
  ({ size, theme }) => `
  display: flex;
  & > div:not(:last-child) {
    margin-right: ${size === 'sm' ? theme.spacing.base.sm : theme.spacing.base.md};
  }
`
)

export const Counter: React.FC<CounterProps> = ({
  days,
  hours,
  minutes,
  seconds,
  maxDaysRange = 365,
  showSeconds = false,
  maxSize = 'lg',
  colors,
  outline,
  lastDayShowSecondsHideDays,
  loading,
  ...otherProps
}): JSX.Element => {
  const { breakpoint: currentBreakpoint } = useWindowDimensions()
  const size = currentBreakpoint < breakpoints.sm ? 'sm' : maxSize

  const showCounters = () => {
    // last day and small device
    if (days === 0 && currentBreakpoint < breakpoints.sm) {
      return { showDaysCounter: false, showSecondsCounter: true }
    }
    const showDaysCounter =
      days > 0 || lastDayShowSecondsHideDays === undefined ? true : !lastDayShowSecondsHideDays
    const showSecondsCounter =
      currentBreakpoint < breakpoints.sm
        ? !showDaysCounter
        : showSeconds || (days === 0 && lastDayShowSecondsHideDays)
    return { showDaysCounter, showSecondsCounter }
  }

  const { showDaysCounter, showSecondsCounter } = showCounters()

  return (
    <StyledCounter size={size} {...otherProps}>
      {showDaysCounter && (
        <CounterItem
          centerValue={days}
          label="Tage"
          maxValue={maxDaysRange}
          size={size}
          colors={colors}
          outline={outline}
          loading={loading}
        />
      )}
      <CounterItem
        centerValue={hours}
        label="Stunden"
        maxValue={24}
        size={size}
        colors={colors}
        outline={outline}
        loading={loading}
      />
      <CounterItem
        centerValue={minutes}
        label="Minuten"
        maxValue={60}
        size={size}
        colors={colors}
        outline={outline}
        loading={loading}
      />
      {showSecondsCounter && (
        <CounterItem
          centerValue={seconds}
          label="Sekunden"
          maxValue={60}
          size={size}
          colors={colors}
          outline={outline}
          loading={loading}
        />
      )}
    </StyledCounter>
  )
}
