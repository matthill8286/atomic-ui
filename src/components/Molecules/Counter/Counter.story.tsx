import { boolean, color, date, number, select } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import React from 'react'
import { Counter } from './Counter'
import { CounterItem } from './CounterItem'
import { CounterWithTimer } from './CounterWithTimer'

const counterKnobs = () => ({
  maxDaysRange: number('Max Days Range', 14),
  maxSize: select('Max Size', ['lg', 'sm'], 'lg'),
  colors: {
    fill: color('Circle Fill Color', '#00000000'),
    fontColor: color('Font Color', '#272422'),
    circleColor: color('Circle Color', '#cfcbca'),
    progressColor: color('Progress Color', '#df0000'),
  },
  outline: boolean('Outline', false),
  backgroundColor: color('Background Color', '#fff'),
  lastDayShowSecondsHideDays: boolean('Hide Days And Show Seconds On Last Day', true),
})

storiesOf('Design System/Molecules/Counter', module)
  .add('Counter With Timer', () => {
    const toDate = date('To', new Date(new Date().getTime() + (86400 * 11 + 6) * 1000))
    const setToUTCToUndefined = boolean('Loading (set "toUTC" to undefined)', false)
    const props = {
      ...counterKnobs(),
      showSeconds: boolean('Show seconds', true),
    }
    return (
      <div style={{ background: props.backgroundColor }}>
        <CounterWithTimer
          toUTC={setToUTCToUndefined ? undefined : new Date(toDate).toUTCString()}
          {...props}
        />
      </div>
    )
  })
  .add('Counter Hides Days and Shows Seconds', () => {
    const toDate = date('To', new Date(new Date().getTime() + (86400 + 3) * 1000))
    const props = {
      ...counterKnobs(),
      showSeconds: boolean('Show seconds', false),
    }
    return (
      <div style={{ background: props.backgroundColor }}>
        <CounterWithTimer toUTC={new Date(toDate).toUTCString()} {...props} />
      </div>
    )
  })
  .add('Counter', () => {
    const days = number('Days', 12)
    const hours = number('Hours', 2)
    const minutes = number('Minutes', 11)
    const seconds = number('Seconds', 59)
    const { maxSize, colors, outline, backgroundColor, maxDaysRange } = counterKnobs()
    const loading = boolean('Loading', false)
    return (
      <div style={{ background: backgroundColor }}>
        <Counter
          days={days}
          hours={hours}
          minutes={minutes}
          seconds={seconds}
          showSeconds
          maxSize={maxSize}
          colors={colors}
          outline={outline}
          maxDaysRange={maxDaysRange}
          loading={loading}
        />
      </div>
    )
  })
  .add('Counter Item', () => {
    const centerValue = number('Center Value', 15)
    const size = select('Size', ['lg', 'sm'], 'lg')
    const fontColor = color('Font Color', '#272422')
    const circleColor = color('Circle Color', '#cfcbca')
    const progressColor = color('Progress Color', '#df0000')
    const outline = boolean('Outline', false)
    const backgroundColor = color('Background Color', '#fff')
    const loading = boolean('Loading', false)
    return (
      <div style={{ background: backgroundColor }}>
        <CounterItem
          label={'Minuten'}
          centerValue={centerValue}
          maxValue={60}
          size={size}
          colors={{ fontColor, circleColor, progressColor }}
          outline={outline}
          loading={loading}
        />
      </div>
    )
  })
