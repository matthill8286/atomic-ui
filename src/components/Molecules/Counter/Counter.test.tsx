import React from 'react'
import { renderWithTheme } from '@/testRenderer'
import { Counter } from './Counter'

describe('Counter', () => {
  it('renders large size', () => {
    const tree = renderWithTheme(
      <Counter
        days={11}
        hours={0}
        minutes={0}
        seconds={59}
        showSeconds={true}
        colors={{ fontColor: '#123456', circleColor: '#345678', progressColor: '#567890' }}
        maxSize="lg"
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders large size with outline', () => {
    const tree = renderWithTheme(
      <Counter
        days={11}
        hours={0}
        minutes={0}
        seconds={59}
        showSeconds={true}
        colors={{ fontColor: '#123456', circleColor: '#345678', progressColor: '#567890' }}
        maxSize="lg"
        outline
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders small size', () => {
    const tree = renderWithTheme(
      <Counter
        days={11}
        hours={0}
        minutes={0}
        seconds={59}
        showSeconds={false}
        colors={{ fontColor: '#123456', circleColor: '#345678', progressColor: '#567890' }}
        maxSize="sm"
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
