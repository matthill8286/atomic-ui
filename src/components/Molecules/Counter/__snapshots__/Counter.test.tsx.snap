// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders large size 1`] = `
.c6 {
  font-family: SourceSansPro,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 a,
.c6 a:focus,
.c6 a:hover,
.c6 a:active,
.c6 a:visited {
  color: inherit;
}

.c1 {
  max-height: 164px;
  min-height: 85px;
  max-width: 164px;
  min-width: 85px;
}

.c2 {
  stroke: #345678;
}

.c3 {
  stroke: #567890;
}

.c4 {
  font-weight: bold;
  font-family: MMHeadline,Arial,sans-serif;
  fill: #123456;
  font-size: 80px;
}

.c5 {
  position: relative;
  text-align: center;
  top: -55px;
}

.c5 span {
  color: #123456;
  font-size: 14px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > div:not(:last-child) {
  margin-right: 24px;
}

<div
  className="c0"
  size="lg"
>
  <div
    className="c1"
    size="lg"
  >
    <svg
      height="100%"
      viewBox="-4 -4 164 164"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={78}
        cy={78}
        fill="none"
        r={78}
        strokeWidth={8}
      />
      <circle
        className="c3"
        color="#567890"
        cx={78}
        cy={78}
        fill="none"
        r={78}
        strokeDasharray={490.0884539600077}
        strokeDashoffset={490}
        strokeLinecap="round"
        strokeWidth={8.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 78 78)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-4}
        size="lg"
        textAnchor="middle"
        x="50%"
        y="60%"
      >
        
        11
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="lg"
    >
      <span
        className="c6"
      >
        Tage
      </span>
    </div>
  </div>
  <div
    className="c1"
    size="lg"
  >
    <svg
      height="100%"
      viewBox="-4 -4 164 164"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={78}
        cy={78}
        fill="none"
        r={78}
        strokeWidth={8}
      />
      <circle
        className="c3"
        color="#567890"
        cx={78}
        cy={78}
        fill="none"
        r={78}
        strokeDasharray={490.0884539600077}
        strokeDashoffset={490}
        strokeLinecap="round"
        strokeWidth={8.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 78 78)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-4}
        size="lg"
        textAnchor="middle"
        x="50%"
        y="60%"
      >
        0
        0
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="lg"
    >
      <span
        className="c6"
      >
        Stunden
      </span>
    </div>
  </div>
  <div
    className="c1"
    size="lg"
  >
    <svg
      height="100%"
      viewBox="-4 -4 164 164"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={78}
        cy={78}
        fill="none"
        r={78}
        strokeWidth={8}
      />
      <circle
        className="c3"
        color="#567890"
        cx={78}
        cy={78}
        fill="none"
        r={78}
        strokeDasharray={490.0884539600077}
        strokeDashoffset={490}
        strokeLinecap="round"
        strokeWidth={8.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 78 78)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-4}
        size="lg"
        textAnchor="middle"
        x="50%"
        y="60%"
      >
        0
        0
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="lg"
    >
      <span
        className="c6"
      >
        Minuten
      </span>
    </div>
  </div>
  <div
    className="c1"
    size="lg"
  >
    <svg
      height="100%"
      viewBox="-4 -4 164 164"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={78}
        cy={78}
        fill="none"
        r={78}
        strokeWidth={8}
      />
      <circle
        className="c3"
        color="#567890"
        cx={78}
        cy={78}
        fill="none"
        r={78}
        strokeDasharray={490.0884539600077}
        strokeDashoffset={490}
        strokeLinecap="round"
        strokeWidth={8.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 78 78)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-4}
        size="lg"
        textAnchor="middle"
        x="50%"
        y="60%"
      >
        
        59
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="lg"
    >
      <span
        className="c6"
      >
        Sekunden
      </span>
    </div>
  </div>
</div>
`;

exports[`Counter renders large size with outline 1`] = `
.c6 {
  font-family: SourceSansPro,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 a,
.c6 a:focus,
.c6 a:hover,
.c6 a:active,
.c6 a:visited {
  color: inherit;
}

.c1 {
  max-height: 164px;
  min-height: 85px;
  max-width: 164px;
  min-width: 85px;
}

.c2 {
  stroke: #345678;
}

.c3 {
  stroke: #567890;
}

.c4 {
  font-weight: bold;
  font-family: MMHeadline,Arial,sans-serif;
  fill: #123456;
  font-size: 80px;
}

.c5 {
  position: relative;
  text-align: center;
  top: -55px;
}

.c5 span {
  color: #123456;
  font-size: 14px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > div:not(:last-child) {
  margin-right: 24px;
}

<div
  className="c0"
  size="lg"
>
  <div
    className="c1"
    size="lg"
  >
    <svg
      height="100%"
      viewBox="-4 -4 164 164"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={78}
        cy={78}
        fill="none"
        r={74}
        strokeWidth={16}
      />
      <circle
        className="c3"
        color="#567890"
        cx={78}
        cy={78}
        fill="none"
        r={74}
        strokeDasharray={464.9557127312894}
        strokeDashoffset={451}
        strokeLinecap="round"
        strokeWidth={8.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 78 78)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-4}
        size="lg"
        textAnchor="middle"
        x="50%"
        y="60%"
      >
        
        11
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="lg"
    >
      <span
        className="c6"
      >
        Tage
      </span>
    </div>
  </div>
  <div
    className="c1"
    size="lg"
  >
    <svg
      height="100%"
      viewBox="-4 -4 164 164"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={78}
        cy={78}
        fill="none"
        r={74}
        strokeWidth={16}
      />
      <circle
        className="c3"
        color="#567890"
        cx={78}
        cy={78}
        fill="none"
        r={74}
        strokeDasharray={464.9557127312894}
        strokeDashoffset={465}
        strokeLinecap="round"
        strokeWidth={8.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 78 78)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-4}
        size="lg"
        textAnchor="middle"
        x="50%"
        y="60%"
      >
        0
        0
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="lg"
    >
      <span
        className="c6"
      >
        Stunden
      </span>
    </div>
  </div>
  <div
    className="c1"
    size="lg"
  >
    <svg
      height="100%"
      viewBox="-4 -4 164 164"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={78}
        cy={78}
        fill="none"
        r={74}
        strokeWidth={16}
      />
      <circle
        className="c3"
        color="#567890"
        cx={78}
        cy={78}
        fill="none"
        r={74}
        strokeDasharray={464.9557127312894}
        strokeDashoffset={465}
        strokeLinecap="round"
        strokeWidth={8.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 78 78)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-4}
        size="lg"
        textAnchor="middle"
        x="50%"
        y="60%"
      >
        0
        0
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="lg"
    >
      <span
        className="c6"
      >
        Minuten
      </span>
    </div>
  </div>
  <div
    className="c1"
    size="lg"
  >
    <svg
      height="100%"
      viewBox="-4 -4 164 164"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={78}
        cy={78}
        fill="none"
        r={74}
        strokeWidth={16}
      />
      <circle
        className="c3"
        color="#567890"
        cx={78}
        cy={78}
        fill="none"
        r={74}
        strokeDasharray={464.9557127312894}
        strokeDashoffset={9}
        strokeLinecap="round"
        strokeWidth={8.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 78 78)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-4}
        size="lg"
        textAnchor="middle"
        x="50%"
        y="60%"
      >
        
        59
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="lg"
    >
      <span
        className="c6"
      >
        Sekunden
      </span>
    </div>
  </div>
</div>
`;

exports[`Counter renders small size 1`] = `
.c6 {
  font-family: SourceSansPro,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 a,
.c6 a:focus,
.c6 a:hover,
.c6 a:active,
.c6 a:visited {
  color: inherit;
}

.c1 {
  max-height: 96px;
  min-height: 85px;
  max-width: 96px;
  min-width: 85px;
}

.c2 {
  stroke: #345678;
}

.c3 {
  stroke: #567890;
}

.c4 {
  font-weight: bold;
  font-family: MMHeadline,Arial,sans-serif;
  fill: #123456;
  font-size: 32px;
}

.c5 {
  position: relative;
  text-align: center;
  top: -48px;
}

.c5 span {
  color: #123456;
  font-size: 12px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > div:not(:last-child) {
  margin-right: 16px;
}

<div
  className="c0"
  size="sm"
>
  <div
    className="c1"
    size="sm"
  >
    <svg
      height="100%"
      viewBox="-2 -2 96 96"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={46}
        cy={46}
        fill="none"
        r={46}
        strokeWidth={4}
      />
      <circle
        className="c3"
        color="#567890"
        cx={46}
        cy={46}
        fill="none"
        r={46}
        strokeDasharray={289.02652413026095}
        strokeDashoffset={289}
        strokeLinecap="round"
        strokeWidth={4.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 46 46)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-2}
        size="sm"
        textAnchor="middle"
        x="50%"
        y="51%"
      >
        
        11
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="sm"
    >
      <span
        className="c6"
      >
        Tage
      </span>
    </div>
  </div>
  <div
    className="c1"
    size="sm"
  >
    <svg
      height="100%"
      viewBox="-2 -2 96 96"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={46}
        cy={46}
        fill="none"
        r={46}
        strokeWidth={4}
      />
      <circle
        className="c3"
        color="#567890"
        cx={46}
        cy={46}
        fill="none"
        r={46}
        strokeDasharray={289.02652413026095}
        strokeDashoffset={289}
        strokeLinecap="round"
        strokeWidth={4.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 46 46)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-2}
        size="sm"
        textAnchor="middle"
        x="50%"
        y="51%"
      >
        0
        0
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="sm"
    >
      <span
        className="c6"
      >
        Stunden
      </span>
    </div>
  </div>
  <div
    className="c1"
    size="sm"
  >
    <svg
      height="100%"
      viewBox="-2 -2 96 96"
      width="100%"
    >
      <circle
        className="c2"
        color="#345678"
        cx={46}
        cy={46}
        fill="none"
        r={46}
        strokeWidth={4}
      />
      <circle
        className="c3"
        color="#567890"
        cx={46}
        cy={46}
        fill="none"
        r={46}
        strokeDasharray={289.02652413026095}
        strokeDashoffset={289}
        strokeLinecap="round"
        strokeWidth={4.5}
        style={
          Object {
            "transition": "stroke-dashoffset 1s ease-out",
          }
        }
        transform="rotate(270 46 46)"
      />
      <text
        className="c4"
        color="#123456"
        dx={-2}
        size="sm"
        textAnchor="middle"
        x="50%"
        y="51%"
      >
        0
        0
      </text>
    </svg>
    <div
      className="c5"
      color="#123456"
      size="sm"
    >
      <span
        className="c6"
      >
        Minuten
      </span>
    </div>
  </div>
</div>
`;
